Kernel/CoreDump.cpp

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (c) 2019-2020, Jesse Buhagiar <jooster669@gmail.com>
+ * Copyright (c) 2020, Itamar S. <itamar8910@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/ByteBuffer.h>
+#include <Kernel/CoreDump.h>
+#include <Kernel/FileSystem/Custody.h>
+#include <Kernel/FileSystem/FileDescription.h>
+#include <Kernel/FileSystem/VirtualFileSystem.h>
+#include <Kernel/Process.h>
+#include <Kernel/Ptrace.h>
+#include <Kernel/RTC.h>
+#include <Kernel/SpinLock.h>
+#include <Kernel/VM/ProcessPagingScope.h>
+#include <LibELF/CoreDump.h>
+#include <LibELF/exec_elf.h>
+
+namespace Kernel {
+
+OwnPtr<CoreDump> CoreDump::create(Process& process)
+{
+    auto fd = create_target_file(process);
+    if (!fd)
+        return nullptr;
+    return make<CoreDump>(process, fd.release_nonnull());
+}
+
+CoreDump::CoreDump(Process& process, NonnullRefPtr<FileDescription>&& fd)
+    : m_process(process)
+    , m_fd(move(fd))
+    , m_num_program_headers(process.m_regions.size() + 1) // +1 for NOTE segment
+{
+}
+
+CoreDump::~CoreDump()
+{
+}
+
+RefPtr<FileDescription> CoreDump::create_target_file(const Process& process)
+{
+    static constexpr const char* coredumps_directory = "/tmp/coredump";
```

Comment:
We should definitely move away from this eventually, so we don't have hardcoded paths in the kernel like this.
However, I think it's okay at the moment.